biogrid/libs/biogrid-simulator/src/lib/community/ruralarea.spec.ts

```diff
@@ -1,30 +1,49 @@
 import { Building } from '../building';
 import { RuralArea } from './';
 
-
 describe('tests for RuralArea class', () => {
+  const x = 10,
+    y = 10;
   test('getBuildingById()', () => {
-    const buildings : Building[] = [new Building(5), new Building(10), new Building(20)];
-    const ruralarea = new RuralArea(buildings);
+    const buildings: Building[] = [
+      new Building(5, 2, 3),
+      new Building(10, 3, 4),
+      new Building(20, 1, 2),
+    ];
+    const ruralarea = new RuralArea(buildings, x, y);
     const aBuildingId = buildings[1].getBuildingId();
     expect(ruralarea.getBuildingById(aBuildingId)).toEqual(buildings[1]);
   });
 
   test('addBuilding()', () => {
-    const buildings : Building[] = [new Building(5), new Building(10), new Building(20)];
-    const ruralarea = new RuralArea(buildings);
-    const newBuilding = new Building(40);
-    ruralarea.addBuilding(newBuilding);
-    expect(ruralarea.getBuildings().length).toEqual(4);
+    const buildings: Building[] = [
+      new Building(5, 2, 3),
+      new Building(10, 3, 4),
+      new Building(20, 1, 2),
+    ];
+    const ruralarea = new RuralArea(buildings, 12, 12);
+    const newBuilding = new Building(40, x, y);
+    ruralarea.addEnergyUser(newBuilding);
+    expect(ruralarea.getEnergyUsers().length).toEqual(4);
   });
-  
-  test('addBuilding() unique Id', () => {
-    const buildings : Building[] = [new Building(5)];
-    const ruralarea = new RuralArea(buildings);
-    const newBuilding = new Building(40);
-    ruralarea.addBuilding(newBuilding);
-    expect(ruralarea.getBuildings()[1].getBuildingId()).toEqual(buildings[0].getBuildingId());
+
+  test('addEnergyUser() unique Id', () => {
+    const buildings: Building[] = [
+      new Building(5, 2, 2),
+      new Building(8, 1, 1),
+    ];
+    const ruralarea = new RuralArea(buildings, x, y);
+    const newBuilding = new Building(40, 9, 9);
+    ruralarea.addEnergyUser(newBuilding);
+    expect(ruralarea.getEnergyUsers()[1].getBuildingId()).toEqual(
+      buildings[1].getBuildingId()
```

Comment:
this doesn't really check that this is a unique ID, this test looks like its just checking the same thing as addBuilding()
Instead, can you make sure no 2 IDs are the same in the list?